Dm 4494 fits header

lznakano opened the pull request:
The flexGrow on the tab pane does not work as I wanted.  Since I am very new to CSS, I did not understand why it did not work.  Please let me know if any of you have a quick answer.  Other that this issue, I believe everything else works.

Please review and let me know your comments.
Many thanks!

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+
+const popupForm  =  (plot, fitsHeaderInfo,  popupId )=> {
+
+	var stats =  fitsHeaderInfo.hasOwnProperty('NO_BAND') ?
```
You don't need to `fitsHeaderInfo.hasOwnProperty('NO_BAND')` just say `fitsHeaderInfo.NO_BAND`

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+			showFitsHeaderPopup(plot, tableId, result);
+		})
+		.catch ( (e) => {
+			console.log(`fitsHeader error: ${plot.plotId}`, e);
```
Use logError in WebUtil

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+	if  (!plot)  return;
+
+	var str=plot.title.replace(/\s/g, '');
+	var tableId= str.replace(/[^a-zA-Z0-9]/g, '_');
```
Nice use of regular expressions!

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+const mTableStyle={width: '100%', height: 380, flexGrow: 1, display: 'flex'};
+
+
+const popupForm  =  (plot, fitsHeaderInfo,  popupId )=> {
```
I don't think you should use the arrow syntax for a long functions.  Just use `function`

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+			 { stats}
+			<div>
+			    <div style={closeButtonStyle}>
+			        <input type='button'  value='close' onClick= {()=>dispatchHideDialog( popupId)} />
```
Use `<CompleteButton  text='close' dialogId='popupId'/>`

lznakano replied:
Good idea.  I will remember to use our components.

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+	callGetFitsHeaderInfo(plot.plotState, tableId)
+		.then( (result) => {
+
+			console.log(result);
```
take this line out.

lznakano replied:
Thank you very much for paying attention to the details.  I forgot to take this debugging line out.

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+ * @constructor
+ */
+export function  FitsHeaderView(plotView){
+
```
This is not a react component. It is just a function. It should be `fitsHeaderView`

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+
+}
+FitsHeaderView.propTypes= {
+	plotView: PropTypes.object.isRequired
```
since this is just a function and not a react component it does not need propTypes.

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+	var popupId=popupIdRoot+'_'+tableId;
+
+	const popTitle = 'FITS Header:'+plot.title;
+	var popup = (<PopupPanel title={popTitle}>
```
This should have spacing. Better to do `FITS Header: ${plot.title}`

robyww commented on src/firefly/js/visualize/ui/FitsHeaderView.jsx:
```diff
@@ -0,0 +1,183 @@
+	const  pixelSize = pt.toFixed(2)+'"';
+	const  meta = tableModel.tableMeta;
+	const  fileSize = Math.round(meta.fileSize/1000.0) +'K';
+
```
This does not scale: it might be megs or gigs. Why don't you go find the gwt code: `FileUtil.getSizeAsString` static method and port it to JS (put it into ).  It is not very big. Put it into WebUtil.js

lznakano replied:
I was looking for java code but did not find it.   I will port the FileUtil.getSizeAsString to the WebUtil.js.

robyww commented on src/firefly/js/core/JsonUtils.js:
```diff
@@ -53,14 +53,21 @@ export const jsonRequest= function(baseUrl, cmd, paramList) {
                     resolve(result[0].data);
                 }
-                else if (result[0].error) {
+                else if (result[0] && result[0].error) {
```
Good place for lodash.get. `get(result,'0.error')`

robyww commented:
UI Notes:
 - bug: sorting hangs, you might need to talk to @loitly 
 - look at the the GWT version of the file size and pixel size line, make it look like that.
 - There is a two scroll bar scrolling problem. You can scroll the table but also scroll the whole component. The whole component should not scroll.
 - For the 3 color I can grow the dialog bigger than the content
 - I think the horizontal scroll bar is showing when it should not as well.
 - some of the sizing might be solved using flex: 
                https://css-tricks.com/snippets/css/a-guide-to-flexbox/
 - I like you use the div resize attribute.  This is not something we do in the GWT version but it is more standard for the web.  I think it is better.
- We have several problems when multiple dialogs are up. This is my issue.  Just made a ticket about it:  DM-5659

robyww commented:
Your react/js code is looking much cleaner.  Good use of functional components and function in general.

Most of the big issues are around making the css behave correctly. I think flex will help with that.  Flex is very good for resize.

lznakano commented:
Hi Trey,

Thank you very much for your kindly reviewing the codes and for your 
great suggestions.
I believe that I updated most of them except two.

1.  When the table put on the tab pane, it does not work the same way as 
it is in the div container.  I did not have the luck to get it work. I 
made the popup size not changeable for now until I learn how to do it.
2. The sorting does not work.  It seems there is some issue in 
TableUtil.js.  I got the same problem when I tried the a table in "Data 
Sets:Catalogs & Images".  I am going to look into it now.

I push my recently changes in.  Please take a look when you have some time.

Thanks,
Lijun

On 3/30/16 9:17 PM, Trey Roby wrote:
>
> UI Notes:
>
>   * bug: sorting hangs, you might need to talk to @loitly
>     <https://github.com/loitly>
>   * look at the the GWT version of the file size and pixel size line,
>     make it look like that.
>   * There is a two scroll bar scrolling problem. You can scroll the
>     table but also scroll the whole component. The whole component
>     should not scroll.
>   * For the 3 color I can grow the dialog bigger than the content
>   * I think the horizontal scroll bar is showing when it should not as
>     well.
>   * some of the sizing might be solved using flex:
>     https://css-tricks.com/snippets/css/a-guide-to-flexbox/
>   * I like you use the div resize attribute. This is not something we
>     do in the GWT version but it is more standard for the web. I think
>     it is better.
>       o We have several problems when multiple dialogs are up. This is
>         my issue. Just made a ticket about it: DM-5659
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/46#issuecomment-203744450>
>

-- 
Lijun Zhang

LSST/IPAC
California Institute of Technology
770 S. Wilson Ave, Mail Code 100-22
Pasadena, CA 91125, USA
Voice: 1-626-395-1972 Fax:1-626-395-1816

robyww commented:
It has improved a lot. I don't have time to go over is throughly today but I did notice a couple of things.
 - The header still does not look like GWT. Look at the one.  The words are bolded and have a little padding.
 - Why did not not allow horizontal resize?  It should be there.

xiuqin commented:
I want the value column to be wider by default.  The button position in the dialog should be consistent. Maybe it should be the job of base dialog? We have some in the middle, some to right currently.

robyww commented:
We just have a CompleteButton component. We might need to have a container that will lay it out like that

xiuqin commented:
I will make a ticket for it.  Nothing need to be done right now. 
Should the resize also in a base dialog? The example dialogs are not resizable.

robyww commented:
I don't think so. I like Lijun's approach of just making a resizable div. Keeps the code more simple.

lznakano commented:
Trey, 
Thank you for pointing out the difference in the text appearance.  I changed the font to black bold.  I accidentally disabled horizontal bar.  I added it back.  Sorting is still not working.  The size in 3-color
popup does not grow vertically. So I set it none-sizable for now until I figure out how to do that.
 
Thanks for your review.
